### PR TITLE
Problem: macos cross compiling failed

### PR DESCRIPTION
**Pull request by mmsqe**
Solution:
- update nixpkgs to master
- update python310

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by yihuang — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ concurrency:
 jobs:
   integration_tests:
     runs-on: ubuntu-latest
-    timeout-minutes: 240
+    timeout-minutes: 480
```
Comment (mmsqe) [inline .github/workflows/test.yml:18]: could revert after this

File: integration_tests/pyproject.toml [hunk 2]
```diff
@@ -13,22 +13,19 @@ black = "^22.3.0"
 flake8-black = "^0.3.2"
 flake8-isort = "^4.1.1"
 pep8-naming = "^0.11.1"
-protobuf = "^3.13.0"
 grpcio = "^1.33.2"
 PyYAML = "^5.3.1"
 python-dateutil = "^2.8.1"
-web3 = "^5.20.1"
-eth-bloom = "^1.0.4"
+web3 = "^6.4.0"
```
Comment (mmsqe) [inline integration_tests/pyproject.toml:19]: seems poetry update can't solve version correctly, while getting no [.egg-info](https://github.com/nix-community/poetry2nix/issues/1161) in newer web3.

Comment (yihuang) [reply integration_tests/pyproject.toml:19]: can we simply use the lock files from main branch?

Comment (mmsqe) [reply integration_tests/pyproject.toml]: same error

Comment (yihuang) [reply integration_tests/pyproject.toml]: Pin to same nixpkgs version?

**Pull request merged**